plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaComputerLauncher.java

```diff
@@ -0,0 +1,56 @@
+package io.jenkins.plugins.remotingkafka;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.TaskListener;
+import hudson.slaves.ComputerLauncher;
+import hudson.slaves.SlaveComputer;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+public class KafkaComputerLauncher extends ComputerLauncher {
+    private static final Logger LOGGER = Logger.getLogger(KafkaComputerLauncher.class.getName());
+
+    @DataBoundConstructor
+    public KafkaComputerLauncher() {
+
+    }
+
+    @Override
+    public boolean isLaunchSupported() {
+        return true;
+    }
+
+    @Override
+    public synchronized void launch(SlaveComputer computer, final TaskListener listener) throws IOException, InterruptedException {
+        KafkaSlaveComputer kafkaComputer = new KafkaSlaveComputer(computer);
```

Comment:
As we discussed on the can yesterday, it is not going to work.
If you want to stick to `option 3` in the ticket, `SlaveComputer computer` should be an instance of `KafkaSlaveComputer`